st2client/st2client/commands/action.py

```diff
@@ -765,9 +768,12 @@ def run(self, args, **kwargs):
         execution.action = action_ref
         execution.parameters = action_parameters
 
-        if args.trace_tag:
+        if not args.trace_id and args.trace_tag:
             execution.context = {'trace_context': {'trace_tag': args.trace_tag}}
 
+        if args.trace_id:
+            execution.context = {'trace_context': {'id_': args.trace_id}}
```

Comment:
One other thing is TraceContext() constructor could fail if people shove extra params in trace_context. So explicilty creating a dict with kwargs we look for is something we should do.